cmd/sandbox-ce/compose/compose.go

```diff
@@ -0,0 +1,134 @@
+package compose
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/mitchellh/go-homedir"
+	"github.com/pkg/errors"
+)
+
+const composeContainerPath = "https://github.com/docker/compose/releases/download/1.24.0/run.sh"
+
+var envKeys = map[string]bool{
+	"GITBASE_REPOS_DIR": true,
+}
+
+type Compose struct {
+	bin string
+}
+
+func (c *Compose) Run(ctx context.Context, arg ...string) error {
+	cmd := exec.CommandContext(ctx, c.bin, arg...)
+
+	var compOpts []string
+	for key := range envKeys {
+		val, ok := os.LookupEnv(key)
+		if !ok {
+			continue
+		}
+
+		compOpts = append(compOpts, fmt.Sprintf("-e %s=%s", key, val))
+	}
+
+	cmd.Env = []string{
+		fmt.Sprintf("COMPOSE_OPTIONS=%s", strings.Join(compOpts, " ")),
+	}
+	cmd.Stdin = os.Stdin
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	return cmd.Run()
+}
+
+func NewCompose() (*Compose, error) {
+	bin, err := getOrInstallComposeBinary()
+	if err != nil {
+		return nil, err
+	}
+
+	return &Compose{bin: bin}, nil
+}
+
+func getOrInstallComposeBinary() (string, error) {
+	path, err := exec.LookPath("docker-compose")
+	if err == nil {
+		bin := strings.TrimSpace(path)
+		if bin != "" {
+			return bin, nil
+		}
+	}
+
+	path, err = getOrInstallComposeContainer()
+	if err != nil {
+		return "", errors.Wrapf(err, "error while getting docker-compose container")
+	}
+
+	return path, nil
+}
+
+func getOrInstallComposeContainer() (string, error) {
+	homedir, err := homedir.Dir()
+	if err != nil {
+		return "", errors.Wrapf(err, "unable to get home dir")
+	}
+
+	dirPath := filepath.Join(homedir, ".srcd", "bin")
+	path := filepath.Join(dirPath, "docker-compose")
```

Comment:
I don't know if it's a problem or not. But if you change the version of the script above it will still use the old one.

Reply:
I think that it's better to keep a fixed version and to update the `composeContainerPath` manually when we need so that we're sure about the expected behavior. On the other hand, we don't have this fully controlled as the user could have its own docker-compose already installed.

So yup, we could eventually always check for update. But in case I'd add this in a second iteration.

Reply:
🤦‍♂ By reading again at this I understood what you mean. I'll add a comment that the version is in the `composeContainerPath` is considered only the first time, but it will be ignored if the script is already present. I'm also gonna open an issue for this.

Reply:
Addressed [here](https://github.com/smacker/superset-compose/pull/4/commits/5159a41e5d0ac7a749a48df7c9672f078d3fcea6).